modules/LoRA.py

```diff
@@ -7,16 +7,29 @@
 from modules.models import reload_model
 
 
-def add_lora_to_model(lora_name):
+def add_lora_to_model(lora_names):
+    prior_set = set(shared.lora_names)
+    added_set = set(lora_names) - prior_set
+    removed_set = prior_set - set(lora_names)
+    shared.lora_names = list(lora_names)
 
-    # If a LoRA had been previously loaded, or if we want
-    # to unload a LoRA, reload the model
-    if shared.lora_name not in ['None', ''] or lora_name in ['None', '']:
-        reload_model()
-    shared.lora_name = lora_name
+    # Nothing to do = skip.
+    if len(added_set) == 0 and len(removed_set) == 0:
+        return
 
-    if lora_name not in ['None', '']:
-        print(f"Adding the LoRA {lora_name} to the model...")
+    # Only adding, and already peft? Do it the easy way.
```

Comment:
Is it correct to assume that the model is already peft? For instance, if you load llama-7b without the `--lora` argument, it will not have been loaded with `PeftModel.from_pretrained`.

Edit: okay, this is only executed if `len(set(shared.lora_names)) > 0`, in which case the model will have been loaded with `PeftModel.from_pretrained`.